packages/datasource-hdt/lib/datasources/ExternalHdtDatasource.js

```diff
@@ -32,10 +33,14 @@ class ExternalHdtDatasource extends Datasource {
     done();
   }
 
+  _matchesGraph(query) {
+    return !query.graph || N3.Util.isDefaultGraph(query.graph) || query.graph.equals(this._graph);
+  }
+
   // Writes the results of the query to the given quad stream
   _executeQuery(query, destination) {
     // Only the default graph has results
-    if (query.graph) {
+    if (!this._matchesGraph(query)) {
```

Comment:
Didn't this work before, because the semantics of this new function seem to be quite different than the logic before.